### PR TITLE
Make Multicombat widgets layoutable

### LINKED ISSUE #12095: Ability to move the 'crossed swords' multicombat icon using Alt clicking
**Is your feature request related to a problem? Please describe.**
Minor issue, but I like to dock my on-screen XP trackers underneath the minimap in resizable mode. When they are attached to the top right anchor point, in multicombat areas, the multicombat icon sits over the top of the onscreen XP tracker, obfuscating the number on the top right.

**Describe the solution you'd like**
Similar to how the minimap and the Runelite canvas elements can be rearranged by holding Alt and dragging them, I would like to see this ability added for the multicombat icon.

**Describe alternatives you've considered**
Note that the XP tracker (or any Alt-moveable UI) can be made to appear in front of the multicombat icon when elements are placed manually underneath the minimap, when it's not connected to the top-right corner anchor. An alternative could be to simply have the multicombat icon sent to the back layer-wise, however I think having the ability to move it would be a nice feature for more client personalisation and customisation.

### PR DESCRIPTION
**Pull request by Hydrox6**
Closes #12095

Resizeable barely changes position by default

Fixed has some minor issues with how it interacts with non-RS moveable widgets, such as Infoboxes
java_2018-10-13_22-55-14

Though it does fit nicely under the XP tracker now

### CODE REVIEW

File: runelite-api/src/main/java/net/runelite/api/widgets/WidgetInfo.java [hunk 1]
```diff
@@ -447,7 +447,10 @@
 	DEADMAN_PROTECTION_TEXT(WidgetID.PVP_GROUP_ID, WidgetID.Pvp.WILDERNESS_LEVEL),
 	DEADMAN_PROTECTION_TIME(WidgetID.PVP_GROUP_ID, WidgetID.Pvp.DEADMAN_PROTECTION_TIME),
 
-	ZEAH_MESS_HALL_COOKING_DISPLAY(WidgetID.ZEAH_MESS_HALL_GROUP_ID, WidgetID.Zeah.MESS_HALL_COOKING_DISPLAY);
+	ZEAH_MESS_HALL_COOKING_DISPLAY(WidgetID.ZEAH_MESS_HALL_GROUP_ID, WidgetID.Zeah.MESS_HALL_COOKING_DISPLAY),
+
+	MULTICOMBAT_FIXED(WidgetID.FIXED_VIEWPORT_GROUP_ID, 20),
```
Comment (deathbeam) [inline runelite-api/src/main/java/net/runelite/api/widgets/WidgetInfo.java]: remove those magic numbers

File: runelite-api/src/main/java/net/runelite/api/widgets/WidgetInfo.java [hunk 2]
```diff
@@ -447,7 +447,10 @@
...
+	MULTICOMBAT_RESIZEABLE(WidgetID.RESIZABLE_VIEWPORT_BOTTOM_LINE_GROUP_ID, 16);
```
Comment (deathbeam) [inline runelite-api/src/main/java/net/runelite/api/widgets/WidgetInfo.java]: same here

**Comment by Hydrox6** [comment]
Added definitions to WidgetID for both widgets, please let me know if they should have gone into a subclass or not.

**Comment by Nightfirecat** [comment]
They should definitely go into a subclass. The idea is that only group IDs are defined at the top class level. Child IDs are always subclassed.

**Comment by dekvall** [comment]
If this were to be revived it would close #12095

**Comment by Hydrox6** [comment]
Wow, it's been a while. 2 year anniversary of this PR in a week!

Merged the commits together, as other recent WidgetOverlay additions have done.
Updated the IDs to work again (both were off slightly due to additions over the past 2 years)

**Pull request merged**